main/observing/observation_run.py

```diff
@@ -630,3 +673,58 @@ def _shutdown_procedure(self, calibration, cooler=True):
             self.take_calibration_images()
         if cooler:
             self.camera.onThread(self.camera.cooler_set, False)
+
+    def threadcheck(self):
+        '''
+        Description
+        ----------
+        Checks to see if self.monitor has raised a crashed thread,
+        Restarts the crashed threads if there are any
+        
+        Returns
+        -------
+        None
+        '''
+        #print('---In thread Check')
+        threadlist = self.monitor.crash_return()
+        if self.monitor.threadcrash == True:
+            #print('Threads have crashed')
+            threadlist = self.monitor.crash_return()
+            for thname in threadlist:
+                self.threadrestart(thname)
```

Comment:
make sure your method calls match the method names?  here you call "threadrestart", but I see it's defined as just "restart"